### PR TITLE
Improve VHPI logging output and some comments

### PR DESCRIPTION
**Pull request by cmarqu**

### CODE REVIEW

**Review by ktbarrett — COMMENTED** [review]
I'm starting to wonder if we should be prefixing everything, or be using a different logger, say `cocotb.gpi.vhpi` that is _under_ `cocotb.gpi`.

**Comment by cmarqu** [comment]
A different logger would certainly be cleaner, but OTOH, this is a low-tech way that works and is well readable in both source code and output, so maybe not worth additional brain cycles.

**Review by ktbarrett — APPROVED** [review]
There is nothing objectional here.

**Comment by cmarqu** [comment]
I'm actually getting compile errors on Mac OS and with clang that I need to investigate... or just remove that code.

File: cocotb/share/lib/vhpi/VhpiCbHdl.cpp [hunk 1]
```diff
@@ -438,13 +438,16 @@ int VhpiCbHdl::arm_callback()
...
             LOG_ERROR("VHPI: Unable to register a callback handle for VHPI type %s(%d)",
-                         m_impl->reason_to_string(cb_data.reason), cb_data.reason);
+                      m_impl->reason_to_string(cb_data.reason), cb_data.reason);
             goto error;
         }
 
         cbState = (vhpiStateT)vhpi_get(vhpiStateP, new_hdl);
-        if (vhpiEnable != cbState) {
-            LOG_ERROR("VHPI: Registered callback isn't enabled! Got %d", cbState);
+        if (vhpiUndefined == cbState) {
```
Comment (ktbarrett) [inline cocotb/share/lib/vhpi/VhpiCbHdl.cpp:446]: I think you need to check this before you cast the result of the call to `vhpi_get` to a `vhpiStateP`. `vhpiUndefined` is a valid result of `vhpi_get`, but it isn't in the `vhpiStateP` enum, hence the warning.

Comment (cmarqu) [reply cocotb/share/lib/vhpi/VhpiCbHdl.cpp]: Ah, thanks for the explanation.
The current code is not exactly nice, is there a way to improve it? Or is all that casting too much for too little gain?
What prompted this is that `vhpiUndefined` is simulator-dependent:
```
#ifndef IUS
#define vhpiUndefined -1
#else
#define vhpiUndefined 1000
#endif
```
But, oh,  I now see that I am casting `vhpiUndefined` wrong (to `unsigned`), but then, `typedef uint32_t vhpiIntT`... what a mess. I guess I'll just revert that part of the PR.

File: cocotb/share/lib/vhpi/VhpiCbHdl.cpp [hunk 2]
```diff
@@ -438,13 +438,17 @@ int VhpiCbHdl::arm_callback()
...
+                      m_impl->reason_to_string(cb_data.reason), cb_data.reason);
             goto error;
         }
 
-        cbState = (vhpiStateT)vhpi_get(vhpiStateP, new_hdl);
-        if (vhpiEnable != cbState) {
-            LOG_ERROR("VHPI: Registered callback isn't enabled! Got %d", cbState);
+        // don't cast to vhpiStateT immediately because vhpiUndefined is not in the enum
+        vhpiIntT cbState_raw = vhpi_get(vhpiStateP, new_hdl);
+        if ((unsigned int)vhpiUndefined == cbState_raw) {
```
Comment (ktbarrett) [inline cocotb/share/lib/vhpi/VhpiCbHdl.cpp:447]:
```suggestion
        if ((vhpiIntT)vhpiUndefined == cbState_raw) {
```

**Review by ktbarrett — APPROVED** [review]

**Pull request merged**